testutils/src/main/java/com/amplifyframework/testutils/Await.java

```diff
@@ -134,28 +134,37 @@ private Await() {}
             @NonNull ResultErrorEmitter<R, E> resultErrorEmitter,
             @NonNull AtomicReference<R> resultContainer,
             @NonNull AtomicReference<E> errorContainer) {
-
-        CountDownLatch latch = new CountDownLatch(1);
-
-        resultErrorEmitter.emitTo(
-            result -> {
-                resultContainer.set(result);
-                latch.countDown();
-            }, error -> {
-                errorContainer.set(error);
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<Throwable> unexpectedErrorContainer = new AtomicReference<>();
+        final Thread thread = new Thread(() -> {
+            try {
+                resultErrorEmitter.emitTo(
+                    result -> {
+                        resultContainer.set(result);
+                        latch.countDown();
+                    }, error -> {
+                        errorContainer.set(error);
+                        latch.countDown();
+                    }
+                );
+            } catch (Throwable unexpectedFailure) {
+                unexpectedErrorContainer.set(unexpectedFailure);
                 latch.countDown();
             }
```

Comment:
would anything ever get thrown directly to this thread?

Reply:
___Not___ if it is working properly. Before, if something was broken, it would throw on the calling thread, here. But since I've moved the call onto a new thread, I don't want the exception to just kill the thread, while the caller is still alive waiting for a response. I want it to kill the _calling_ thread, too.

But yea - this is a safeguard. Software that is functioning as we intent it do would not trigger this condition. (But, this is after-all a _testutils_ component, so we can't expect things passed into this method will always be functioning correctly (they may throw instead of calling error consumer, accidentally.))

Reply:
I guess it's always better safe than sorry. I just couldn't think of a scenario where anything would ever be thrown directly here now that you created a new well-contained thread.